### PR TITLE
Cloud: CMEK support

### PR DESCRIPTION
**Pull request by filiphaftek**
CMEK documentation - [OKR](https://www.appsheet.com/start/ce3ca2b6-cf8f-4367-a386-28a3e9856cb9#control=Objectives_Detail&row=9O5IZHDIldP8OJhlILAoIJ)

### CODE REVIEW

**Review by daxmc99 — APPROVED** [review]

File: content/departments/cloud/technical-docs/cmek.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Customer Managed Encryption Keys
+
+Customer Managed Encryption Keys (CMEK) is the security feature which ensures that the customer data in Google Cloud Platform are encrypted at rest using unique, dedicated keys stored in Cloud KMS - [Google Cloud Platform documentation](https://cloud.google.com/kms/docs/cmek).
```
Comment (evict) [inline content/departments/cloud/technical-docs/cmek.md]:
```suggestion
Customer Managed Encryption Keys (CMEK) is the security feature which ensures that the customer data in Google Cloud Platform is encrypted at rest using unique, dedicated keys stored in Cloud KMS - [Google Cloud Platform documentation](https://cloud.google.com/kms/docs/cmek).
```

File: content/departments/cloud/technical-docs/cmek.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+Benefits:
+
+- customer's data are encrypted with non-default Google key, which ensures Google (nor any other entity) cannot decrypt the data
```
Comment (evict) [inline content/departments/cloud/technical-docs/cmek.md]:
```suggestion
- customer data is encrypted using a non-default key not provided by Google, ensuring that neither Google nor any other entity can decrypt the data.
```

File: content/departments/cloud/technical-docs/cmek.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+- database data are encrypted with different key than GKE persistent volumes to increase security
```
Comment (evict) [inline content/departments/cloud/technical-docs/cmek.md]:
```suggestion
- database data is encrypted with a different key as GKE persistent volumes, to increase security
```

File: content/departments/cloud/technical-docs/cmek.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+- dedicated CMEK can be rotated manually or automatilly if required
```
Comment (evict) [inline content/departments/cloud/technical-docs/cmek.md]:
```suggestion
- dedicated CMEK can be rotated manually or automatically when required
```

File: content/departments/cloud/technical-docs/cmek.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+## How to enable CMEK
+
+> [!WARNING] CMEK can only be enabled during instance creation. Changing this conguration after the instance is created will require recreation of CloudSQL and Persistent Volumes, which will destroy all the data!
```
Comment (evict) [inline content/departments/cloud/technical-docs/cmek.md]:
```suggestion
> [!WARNING] CMEK can only be enabled during instance creation. Changing this configuration after the instance is created will require recreation of CloudSQL and Persistent Volumes, which will result in a loss of all data
```

File: content/departments/cloud/technical-docs/cmek.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+modify instance `config.yaml` with given annotation:
+
+```yaml
+"cloud.sourcegraph.com/enable-cmek" = "true"
+```
+
+## How to disable CMEK
+
+When CMEK is enabled for instance, it cannot be disabled. Changing this conguration after the instance is created will require recreation of CloudSQL and Persistent Volumes, which will destroy all the data!
```
Comment (evict) [inline content/departments/cloud/technical-docs/cmek.md]:
```suggestion
When CMEK is enabled for instances, it cannot be disabled. Changing this configuration after the instance is created will require recreation of CloudSQL and Persistent Volumes, which will result in a loss of all data.
```

**Review by evict — APPROVED** [review]

**Comment by evict** [comment]
Some textual changes, but overall looks good! 👍

**Comment by filiphaftek** [comment]
@michaellzc do we abandon this PR for Notion change you made?

**Pull request closed without merging**